test/lib/jdk/test/lib/net/JWebServerLauncher.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.test.lib.net;
+
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.process.ProcessTools;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Utility to launch the {@code jwebserver}. This does the necessary work of finding the tool
+ * location and launching the process and waiting for the server to be ready to serve requests.
+ */
+public class JWebServerLauncher {
+
+    private static final Path JWEBSERVER_TOOL = Path.of(JDKToolFinder.getJDKTool("jwebserver"));
+    private static final String EXPECTED_SERVER_STARTUP_LINE = "URL ";
+
+    /**
+     * Represents a launched jwebserver
+     *
+     * @param process       The jwebserver process
+     * @param serverAddr    The address on which the server is listening
+     * @param processOutput The server's stdout and stderr output
+     */
+    public record JWebServerProcess(Process process, InetSocketAddress serverAddr,
```

Comment:
Maybe JWebServerProcess could be AutoCloseable (where close() would destroy the process), which would make it possible to use it in a try-with-resource block?